src/ui/widgets/detail/DetailWidgetElement.js

```diff
@@ -0,0 +1,88 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Icon, DropdownKebab } from 'patternfly-react';
+import { LinkMenuItem } from 'frontend-common-components';
+import { ROUTE_PAGE_DETAIL, ROUTE_PAGE_CONFIG } from 'app-init/router';
+
+const DetailWidgetElement = ({ widgetInfo }) => {
+  const renderWidgetInfo = () => {
+    const { data } = widgetInfo;
```

Comment:
Why not passing data directly as a prop? This way the component has to know about the prop structure...